src/ProtectedRoute.tsx

```diff
@@ -0,0 +1,9 @@
+import React from "react";
+import { Navigate, Outlet } from "react-router-dom";
+
+const ProtectedRoute = ({ children, ...rest }: any) => {
+  let token = localStorage.getItem("token");
```

Comment:
This is not a very good approach. Any one can set anything in the local storage as the token and access your page. Use `isAuth` from the state instead. Also, put a `!loading` check so it doesn’t redirect while loading.